pkg/app/piped/planner/kubernetes/kubernetes.go

```diff
@@ -480,3 +492,51 @@ func determineVersion(manifests []provider.Manifest) (string, error) {
 
 	return b.String(), nil
 }
+
+// determineVersions decides artifact versions of an application.
+// It finds all container images that are being specified in the workload manifests then returns their names, version numbers, and urls.
+func determineVersions(manifests []provider.Manifest) ([]*model.ArtifactVersion, error) {
+	versions := []*model.ArtifactVersion{}
+	imageMap := map[string]string{}
+
+	for _, m := range manifests {
+		if !m.Key.IsDeployment() {
+			continue
+		}
+		data, err := m.MarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+		var d resource.Deployment
+		if err := json.Unmarshal(data, &d); err != nil {
+			return nil, err
+		}
+
+		containers := d.Spec.Template.Spec.Containers
+		// remove duplicate images on multiple manifests
+		for _, c := range containers {
+			imageMap[c.Image] = ""
+		}
+	}
+
+	for i := range imageMap {
+		image := parseContainerImage(i)
+		versions = append(versions, &model.ArtifactVersion{
+			Kind:    model.ArtifactVersion_CONTAINER_IMAGE,
+			Version: image.tag,
+			Name:    image.name,
+			Url:     i,
+		})
+	}
+
+	if len(versions) == 0 {
+		return []*model.ArtifactVersion{
+			{
+				Kind:    model.ArtifactVersion_UNKNOWN,
+				Version: versionUnknown,
+			},
+		}, nil
+	}
```

Comment:
We don't need this part. An empty list is fine. It means that this application is not having any artifact version.